src/presentation/http/router/note.test.ts

```diff
@@ -1792,4 +1861,232 @@ describe('Note API', () => {
+       */
+      let userAccessToken: string = '';
+
+      /**
+       * New note posted by creator
+       */
+      let response = await global.api?.fakeRequest({
+        method: 'POST',
+        headers: {
+          authorization: `Bearer ${creatorAccessToken}`,
+        },
+        body: {
+          content: DEFAULT_NOTE_CONTENT,
+          tools: [
+            {
+              name: headerTool.name,
+              id: headerTool.id,
+            },
+            {
+              name: listTool.name,
+              id: listTool.id,
+            },
+          ],
+        },
+        url: `/note`,
+      });
+
+      /**
+       * Id of the note posted by creator
+       */
+      const noteId = response?.json().id;
+
+      if (authorized) {
+        userAccessToken = global.auth(user.id);
+      }
+
+      if (userCanEdit) {
+        await global.db.insertNoteTeam({
+          noteId: 1,
+          userId: user.id,
+          role: MemberRole.Write,
+        });
+      }
+
+      response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${creatorAccessToken}`,
+        },
+        body: {
+          content: newNoteContent,
+          tools: [
+            {
+              name: headerTool.name,
+              id: headerTool.id,
+            },
+            {
+              name: listTool.name,
+              id: listTool.id,
+            },
+          ],
+        },
+        url: `/note/${noteId}`,
+      });
+
+      /**
+       * Get note history
+       */
+      response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${userAccessToken}`,
+        },
+        url: `/note/${noteId}/history`,
+      });
+
+      if (expectedMessage !== null) {
+        expect(response?.json()).toStrictEqual({ message: expectedMessage });
+      } else if (expectedResponseLength !== undefined) {
+        expect(response?.json().noteHistoryMeta).toHaveLength(expectedResponseLength);
+      }
+    });
+  });
+
+  describe('GET /note/:notePublicId/history', () => {
+    test.each([
+      /**
+       * User can not edit the note state
+       * Should return permission denied response
+       */
+      {
+        authorized: true,
+        userCanEdit: false,
+        expectedMessage: 'Permission denied',
+      },
+      /**
+       * Unauthorized state
+       * Should return unauthorized response
+       */
+      {
+        authorized: false,
+        userCanEdit: true,
+        expectedMessage: 'You must be authenticated to access this resource',
+      },
+      /**
+       * User is authorized and can edit the note
+       * Should return history record that is inserted
+       */
+      {
+        authorized: true,
+        userCanEdit: true,
+        expectedMessage: null,
+      },
+    ])('Should return note history', async ({ authorized, userCanEdit, expectedMessage }) => {
```

Comment:
what is the difference between previous one?

Reply:
updated description